crates/ty_server/src/server.rs

```diff
@@ -35,7 +36,11 @@ pub(crate) struct Server {
 }
 
 impl Server {
-    pub(crate) fn new(worker_threads: NonZeroUsize, connection: Connection) -> crate::Result<Self> {
+    pub(crate) fn new(
+        worker_threads: NonZeroUsize,
+        connection: Connection,
+        fallback_system: Option<Arc<dyn System + 'static + Send + Sync + RefUnwindSafe>>,
```

Comment:
I would probably remove the `Option` here. It seems to complicate things unnecessarily only so that we can fallback to `OsSystem`. Doingt hsi fallback in `Server::new` seems easy enough.

Reply:
Yes, that's what I meant. Creating it in `run_server` seems fine to me. It's similar to what we do in the CLI where the system is one of the first things we create. Unless we can't do that because we need to use a different current directory.